internal/impl/devbox.go

```diff
@@ -394,20 +397,27 @@ func (d *Devbox) GenerateEnvrc(force bool, source string) error {
 			}
 		}
 
-		if strings.ToLower(result) == "y" || !isInteractiveMode {
+		if strings.ToLower(result) == "y" || !isInteractiveMode || source == "generate" {
+			nixDaemon := false
+			if err := nix.EnsureNixInstalled(d.writer, &nixDaemon); err != nil {
+				return err
+			}
+
+			// generate all shell files to ensure we can refer to them in the .envrc script
+			if err := d.ensurePackagesAreInstalled(ctx, ensure); err != nil {
```

Comment:
this is basically a separate bugfix, right?  before this, we could've generated `.envrc` files that referred to non-existent `.nix` files. No we don't 👍

Reply:
yup

The envrc has `if [ -f .devbox/gen/flake/flake.nix]; then` so it wouldn't have errored but would've been a no-op. 

Now, we know it'll actually run the `devbox shellenv --init-hook`.